shell/platform/embedder/embedder.h

```diff
@@ -963,6 +963,52 @@ typedef const FlutterLocale* (*FlutterComputePlatformResolvedLocaleCallback)(
     const FlutterLocale** /* supported_locales*/,
     size_t /* Number of locales*/);
 
+/// Display refers to a graphics hardware system consisting of a framebuffer,
+/// typically a monitor or a screen. This ID is unique per display and is
+/// stable until the Flutter application restarts.
+typedef uint64_t FlutterEngineDisplayId;
+
+typedef struct {
+  /// This size of this struct. Must be sizeof(FlutterDisplay).
+  size_t struct_size;
+
+  FlutterEngineDisplayId display_id;
+
+  /// A double-precision floating-point value representing the actual refresh
+  /// period in seconds. This value may be zero if the device is not running or
+  /// unavaliable or unknown.
+  double refresh_rate;
+
+} FlutterEngineDisplay;
+
+/// The update type parameter that is passed to
+/// `FlutterEngineNotifyDisplayUpdate`.
+typedef enum {
+  /// `FlutterEngineDisplay`s that were active during start-up. A display is
+  /// considered active if:
+  ///    1. The frame buffer hardware is connected.
+  ///    2. The display is drawable, e.g. it isn't being mirrored from another
+  ///    connected display or sleeping.
+  kFlutterEngineDisplaysUpdateTypeStartup,
+  kFlutterEngineDisplaysUpdateTypeCount,
+} FlutterEngineDisplaysUpdateType;
```

Comment:
What's the use case for the update type? Is the idea that updates would be deltas rather than just sending the new list?

If so, doesn't that push unnecessary state-tracking information to every embedding, compared to just having the embeddings send a complete display list after any changes?

Reply:
I was modeling this based on the way platforms seemed to expose the change events, I don't think the embedders need to do a lot of state tracking, the platform apis seem to expose the change events in a similar way:

- On Windows: https://docs.microsoft.com/en-us/uwp/api/windows.devices.display.core.displaymanager.changed?view=winrt-19041
- On macOS: https://developer.apple.com/documentation/coregraphics/cgdisplayreconfigurationcallback?language=objc

I was hoping that would make it easier for the embedders to pipe through the events.